homeassistant/components/prosegur/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow for Prosegur Alarm integration."""
+import logging
+
+from pyprosegur.auth import COUNTRY, Auth
+from pyprosegur.installation import Installation
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {"username": str, "password": str, "country": vol.In(COUNTRY.keys())}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        auth = Auth(session, data["username"], data["password"], data["country"])
+        install = await Installation.retrieve(auth)
+    except ConnectionRefusedError:
+        raise InvalidAuth from ConnectionRefusedError
+
+    # Return info that you want to store in the config entry.
+    return {
+        "title": f"Contract {install.contract}",
+        "contract": install.contract,
+        "username": data["username"],
+        "password": data["password"],
+        "country": data["country"],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Prosegur Alarm."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is None:
```

Comment:
Can you set up multiple Prosegur alarm systems with the same account? If not, you could add something like:

```
await self.async_set_unique_id(user_input.get(CONF_USERNAME))
self._abort_if_unique_id_configured()
```

or use the contract as a unique key, I don't know about the best unique id in your situation.

Reply:
I can theoretically setup multiple alarm systems, unfortunately I don't have many to validate that 😬

Reply:
@dgomes but could you do this with the same credentials as well? The code I shared above will make sure users can't add the same account twice.

Reply:
Theoretically the same credentials can give you access to multiple alarm Installations, since I only have one, I'm currently returning the first.

I except someone with multiple alarms will eventually fill an issue 😄 then I can debug and support more then the first installation.

Reply:
I decided in https://github.com/home-assistant/core/pull/44679/commits/b9546521ddc115f6a9705e5e98b08afe9ef8001f to use the contract, at least it will abort before creating the `config_entry`

Reply:
> Theoretically the same credentials can give you access to multiple alarm Installations, since I only have one, I'm currently returning the first.

True, but you could eventually loop through all available devices and add these entities. It doesn't require another config entry.

Reply:
I'm not certain about the contract being one or multiple... its really a grey area for me...